### PR TITLE
generalizes the state around modals and the modal handler component,

### LINKED ISSUE #840: Help Modal UI is broken
Help Modal's UI is broken

### What I Did
Opened Help Modal

### What I Expected
A scrollable list of keybindings

### What Happened 
No scrollable list. Items are out of the box. The console also has an error specifying invalid DOM String.
screenshot_2018-09-05 notebook 101 - iodide

### (OPTIONAL) notebook that reproduces bug 

https://extremely-alpha.iodide.io/notebooks/1/

### PR DESCRIPTION
**Pull request by bcolloran**
also fixes #840 and clears a warning

@hamilton please review to make sure the code looks reasonable. @mdboom please review to take a look at this pattern to assess whether you want to use this in what you're working on.

(regarding that latter bit: I wonder if it might make sense to invert control here a little bit? -- maybe replace saveNotebookToServer with a wrapper trySavingNotebookToServer that can detect (a) if a user is logged in, and (b) if a user is looking at their own notebook, inwhich case saveNotebookToServer can be called straight away; but if those things are not true, the user can be prompted to log in and/or make a copy)

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ import './style/header-bar-styles.css'
 import './style/side-panes.css'
 import './style/menu-and-button-and-ui-styles.css'
 import './style/cell-styles.css'
+import './style/help-modal-styles.css'
```
Comment (bcolloran) [inline src/index.jsx]: this line just fixes #840

File: src/components/modals/keyboard-shortcut-list.jsx [hunk 2]
```diff
@@ -27,10 +27,10 @@ export default class KeyboardShortcutList extends React.Component {
     return (
       <div className="help-modal-contents">
         <h2>Global keys</h2>
-        <table className="keyboard-shortcuts-global">{globalKeys}</table>
+        <table className="keyboard-shortcuts-global"><tbody>{globalKeys}</tbody></table>
         <h2>Command mode keys (press <pre className="key-combo-pill">Esc</pre> to enter command mode)
         </h2>
-        <table className="keyboard-shortcuts-command-mode">{commandModeKeys}</table>
+        <table className="keyboard-shortcuts-command-mode"><tbody>{commandModeKeys}</tbody></table>
```
Comment (bcolloran) [inline src/components/modals/keyboard-shortcut-list.jsx]: stuff with tbody fixes a warning

File: src/components/modals/__tests__/keyboard-shortcut-list-test.js [hunk 3]
```diff
@@ -56,12 +54,16 @@ describe('HelpModalUnconnected React component', () => {
...
+    expect(getTestComponent()
+      .find('table.keyboard-shortcuts-global tbody')
+      .children().length)
       .toEqual(3)
   })
 
   it('has the right number of command mode shortcuts', () => {
-    expect(getTestComponent().find('.keyboard-shortcuts-command-mode').children().length)
+    expect(getTestComponent()
+      .find('table.keyboard-shortcuts-command-mode tbody')
```
Comment (bcolloran) [inline src/components/modals/__tests__/keyboard-shortcut-list-test.js]: stuff with tbody fixes a warning; this is a related test fix

**Comment by mdboom** [comment]
I'm flummoxed by how this would work.  Is it possible to create something that boiled down to a function call?

**Pull request merged**